verdictd/app/src/protocol.rs

```diff
@@ -0,0 +1,87 @@
+
+use serde_json::Value;
+use crate::key_manager;
+use crate::crypto::aes256_cbc;
+
+fn handle_version() -> Result<String, String> {
+    let mut response = serde_json::Map::new();
+    response.insert("status".to_string(), Value::String("OK".to_string()));
+    response.insert("version".to_string(), Value::String("v1".to_string()));
+
+    Ok(Value::Object(response).to_string())
+}
+
+fn handle_decrypt(request: &Value) -> Result<String, String> {
+    let blobs = match request["blobs"].as_array() {
+        Some(blobs) => blobs,
+        None => return Err("decrypt parameters error".to_string()),
+    };
+
+    let mut response = serde_json::Map::new();
+    response.insert("status".to_string(), Value::String("OK".to_string()));    
+
+    for blob in blobs {
+        if blob["algorithm"] != "AES" || blob["key length"] != 256 || 
+          blob["encrypted data"].is_null() || blob["iv"].is_null() {
+            return Err("parameters error".to_string());
+        }    
+        
+        match key_manager::directory_key_manager::get_key(&String::from(blob["kid"].as_str().unwrap()))
+            .map_err(|_| format!("kid: {}'s key not found", blob["kid"].to_string()))
+            .and_then(|key|{
+                let iv = serde_json::from_str::<Vec<u8>>(blob["iv"].as_str().unwrap()).unwrap();
+                let encrypted_data = serde_json::from_str::<Vec<u8>>(blob["encrypted data"].as_str().unwrap()).unwrap();
```

Comment:
The encrypted data sent by AA should be Base64 encoded, so do you need to add a Base64 decoding step here?

Reply:
Does this data must be sent with Base64 format?